### ISSUE TITLE
Load Saves from S3

### ISSUE THREAD

**Issue by viperior** [stone-tech-inc/rimhistory#15]
Add a feature that allows `rimhistory` to load RimWorld save files hosted in an S3 bucket.

New options in `config.json`:
- `rimworld_save_file_source` (str): The source to use to load the save files (local, s3)
- `rimworld_save_file_s3_bucket` (str): The name of the S3 bucket to load save files from

The current implementation uses `glob` and regex to scan a local directory for save files. It does *not* sniff the XML before loading into a series, although that would be a nice addition to enhance series loading. Instead, it relies on file naming conventions. It ignores autosaves. Rimhistory would work best when paired with a mod that uses templates to name autosaves. I found 2 or 3 such mods available on the Steam Workshop.

To support S3 loading, new functionality must be added to perform the matching against S3 object names in the bucket. After that, the content is loaded as a string into a `Save` object via `SaveSeries`.

**Comment by viperior** [comment]
`Save` objects are currently created with the required argument, `path_to_save_file`:

```python
Save(path_to_save_file="path/to/save.rws")`
```

The path is passed to `ElementTree.parse(path_value)` to load the XML tree into each `Save` object during processing. To support S3 loading, take the extra step of loading the S3 object data into a string variable. Then, pass the XML string to `ElementTree.parse(xml_string)`.

Modify the `Save` class:

1. Create `Save.parse()`, which accepts _either_ a path to a save file or a string containing valid XML.

Modify the `SaveSeries` class:

1. Read the value set in `config.json:rimworld_save_file_source` and handle the first two supported modes: local, s3.
2. If mode == local, use the `rimworld_save_file_dir` and `rimworld_save_file_series_pattern` values to identify the source files.
3. If mode == s3, use the `rimworld_save_file_s3_bucket` and `rimworld_save_file_series_pattern` values to identify the source files.

**Comment by viperior** [comment]
After digging into this task more, I ended up deciding to go in a simpler direction. I identified that the core need is just having multiple save files in a series available for testing. S3 operations might be interesting to support save backups, but that's not the main focus of rimhistory.

Now, the `Save` class will automatically decompress files compressed with `gzip` and having the `.gz` file extension. I took three save files created from a modded playthrough, compressed them with `gzip`, and added them to a new `test_data` directory. Since they are compressed, git tracks them as binary content, which is preferred.

The end result is a more streamlined test suite with a reliable series of save files to use for testing multi-file analytics features.
